utilities/feature_space_measures.py

```diff
@@ -0,0 +1,354 @@
+import numpy as np
+from utilities.scalers import NormalizeScaler, KernelNormalizer
+
+# The code is based on methods presented in the paper TODO(currently in
+# review process, will be added as soon as published)
+
+# default cross validation parameters used for the paper
+DEFAULT_CROSS_VALIDATION_KWARGS = {
+    "train_test_split": True, "train_ratio": 0.5, "seed": 0x5F3759DF, "nb_folds": 2}
+
+
+def standardize_features(features, train_idx=None):
+    """
+    Standardizes the features according to mean and variance of training set
+
+    Parameters:
+    ----------
+    features : array_like
+        features, dimensions: nb samples x nb features
+    train_idx : array_like
+        the training samples are only used to determine mean and variance
+
+    Returns:
+    --------
+    standardized_features : array_like
+        standardized features
+    """
+    if train_idx is None:
+        return NormalizeScaler().fit(features).transform(features)
+    return NormalizeScaler().fit(features[train_idx]).transform(features)
+
+
+def standardize_kernel(kernel, train_idx=None):
+    """
+    Standardizes the kernel according to mean and variance of training set
+
+    Parameters:
+    ----------
+    kernel: array_like
+        features, dimensions: nb samples x nb samples
+    train_idx : array_like
+        the training samples are only used to determine mean and variance
+
+    Returns:
+    --------
+    standardized_kernel : array_like
+        standardized kernel
+    """
+    if train_idx is None:
+        KernelNormalizer().fit(kernel).transform(kernel)
+    return KernelNormalizer().fit(kernel[train_idx, :][:, train_idx]).transform(
+        kernel, K_Atrain_and_Btrain=(kernel[:, train_idx], kernel[:, train_idx].T))
+
+
+def compute_reconstruction_matrix(features, features_dash, nb_folds):
```

Comment:
This feels like a validation wrapper, which should be a post-processing tool, rather than a core tool, no?